pages/library/[libGameId]/index.jsx

```diff
@@ -3,11 +3,39 @@ import BackButton from "@/components/buttons/BackButton";
 import { useRouter } from "next/router";
 import useSWR from "swr";
 import Image from "next/image";
+import {
+  StyledButton,
+  StyledButtonWrapper,
+} from "@/components/buttons/DefaultButtons";
+import MenuButton from "@/components/buttons/MenuButton";
+import { useState, useEffect, useRef } from "react";
+import MenuOption from "@/components/MenuOption";
 
 const fetcher = (...args) => fetch(...args).then((response) => response.json());
-
-export default function LibraryGameDetails() {
+export default function LibraryGameDetails({
+  setGames,
+  games,
+  setMenuMode,
+  menuMode,
+}) {
   const router = useRouter();
+  const [buttonMode, setButtonMode] = useState("default");
+  const dialogRef = useRef(null);
+
+  function gameExists(newGame) {
+    return games?.some((game) => Number(game.id) === newGame[0].id);
+  }
+
+  useEffect(() => {
+    if (buttonMode === "add") {
+      setTimeout(() => {
+        dialogRef.current?.scrollIntoView({
+          behavior: "smooth",
+          block: "center",
+        });
+      }, 50);
+    }
+  }, [buttonMode]);
 
   const { data, error, isLoading } = useSWR(
```

Comment:
This data is an array with only one object, which forces you to write in several instances `data[0]` to access the object itself.

I would add two lines of code below the if-statement where you check whether an error occurred:

```js
if (!data) return <div>Could not find game</div>; // This could of course be some other element and/or text.

  const singleGame = data[0];
```

After that, you can always access the object directly.

This will mean easier to read code, once you changed this for all instances.